internal/ybdata/ybdata.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 YourBase Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// Package ybdata locates directories the user has designated or conventionally
+// uses for storing different types of data.
+package ybdata
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"go4.org/xdgdir"
+)
+
+// Dirs is the result of locating directories.
+type Dirs struct {
+	cache      string
+	workspaces string
+}
```

Comment:
Any particular reason we are passing around `*Dirs` everywhere instead of `Dirs`? It's a fairly light struct and I don't see any case where it's intended to be nil, so would it be worth it to gain the safety of not dealing with pointers?